PennyPincher.API/PennyPincher/Controllers/MoneyManagementController.cs

```diff
@@ -1,76 +1,175 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using PennyPincher.Models;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
 using System.Text.Json;
 
 namespace PennyPincher.Controllers
 {
     [ApiController]
     [Route("/api/management")]
 
-    
+    // This controller is meant to model the projected cash flows you would like to acheive given incomes and expenses
+    // Model hypotheticals or goals and how they align with your current rate of spending in LogHistoryController
     public class MoneyManagementController : Controller
     {
-        //defining projected cash flow
-        static double income = 0;
-        static double expense_food = 0;
-        static double expense_living = 0;
-        static double expense_personal = 0;
+        List<CashFlowDto> CFList = CashFlowDataStore.CurrentCashFlow.CashFlowsList;
+
+        [HttpPost]
+        public ActionResult<CashFlowDto> CreateFlow(CashFlowDto cashFlow) {
+            CFList.Add(cashFlow);
+            return Ok(cashFlow);
+        }
+
+        [HttpGet]
+        public ActionResult<List<CashFlowDto>> GetAllFlows(FlowTypes? targetFlowType) 
+        {
+
+            List<CashFlowDto> CFFiltered= new List<CashFlowDto>();
+
+            if (targetFlowType != null)
+            {
+                CFFiltered = CFList.Where(c => c.Flow.Equals(targetFlowType)).ToList();
+
+                if (CFFiltered.Count <= 0)
+                {
+                    return NotFound("No items were found for this CashFlow type");
+                }
+                return Ok(CFFiltered);
+            }
+            else
+            {
+                return Ok(CashFlowDataStore.CurrentCashFlow.CashFlowsList);
+            }
+        }
 
 
 
-        [HttpGet("income")]
-        public double getIncome()
+        [HttpGet("{targetCashFlowID}", Name = "GetFlow")]
+        public ActionResult<CashFlowDto> GetFlow(int targetCashFlowID)
         {
-            return income;
+            if (CFList.Count() == 0)
+            {
+                return NotFound("No items in Cash Flow");
+            }
+            int pos = CFList.FindIndex(f => f.Id == targetCashFlowID);
 
+            if (pos >= 0)
+            {
+                return Ok(CFList[pos]);
```

Comment:
This totally works, but I can also see a more C# implementation using LINQ and the `.Where` method here to also get the specific cashflow